src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,204 @@
+     */
+    public static void addTask(String line) throws EmptyCommandException, IllegalCommandException, IOException {
+
+        if (!line.startsWith("todo") && !line.startsWith("deadline") && !line.startsWith("event")) {
+            throw new IllegalCommandException();
+        }
+
+        if (line.split(" ").length < 2) {
+            throw new EmptyCommandException();
+        }
+
+        if (line.startsWith("todo")) {
+            tasks.add(Task.getTaskCount(), new Todo(line.replace("todo ", "")));
+        } else if (!line.contains("/")) {
+            throw new IllegalCommandException();
+        } else if (line.startsWith("deadline")) {
+            String[] words = line.split(" /by ");
+            tasks.add(Task.getTaskCount(), new Deadline(words[0].replace("deadline ", ""), words[1]));
+        } else if (line.startsWith("event")) {
+            String[] words = line.split(" /at ");
+            tasks.add(Task.getTaskCount(), new Event(words[0].replace("event ", ""), words[1]));
+        }
+
+        System.out.println("I've added that to your list:\n" + tasks.get(Task.getTaskCount() - 1));
+        System.out.println("Now you have " + Task.getTaskCount() + " tasks in the list.");
+        saveData();
+    }
+
+    /**
+     * Sets a task in the list (specified by the user) as done.
+     * The command is entered in the format "done X", where X is
+     * the task's number.
+     *
+     * @param line The command entered by the user.
+     */
+    public static void doneTask(String line) throws EmptyCommandException, IllegalCommandException, IOException {
+        String[] words = line.split(" ");
+        if (words.length < 2) {
+            throw new EmptyCommandException();
+        } else {
+            try {
+                int taskIndex = Integer.parseInt(words[1]);
+                tasks.get(taskIndex - 1).setDone(true);
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println(tasks.get(taskIndex - 1));
+            } catch (NumberFormatException e) {
+                throw new IllegalCommandException();
+            }
+        }
+        saveData();
+    }
+
+    public static void deleteTask(String line) throws EmptyCommandException, IllegalCommandException, IOException {
+        String[] words = line.split(" ");
+        if (words.length < 2) {
+            throw new EmptyCommandException();
+        } else {
+            try {
+                int taskIndex = Integer.parseInt(words[1]);
+                System.out.println("Okay, I've deleted that task!");
+                System.out.println(tasks.get(taskIndex - 1));
+                tasks.remove(taskIndex - 1);
+            } catch (NumberFormatException e) {
+                throw new IllegalCommandException();
+            }
+        }
+        saveData();
+    }
+
+    public static void startDuke() throws IOException {
+        if (Files.notExists(Paths.get(FILE_PATH))) {
```

Comment:
Similar to before, can make the happy path clearer. You can `return` after you create the file and directories. Then the rest of the logic doesn't need to be nested in `else`.